spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseBatchReader.java

```diff
@@ -42,11 +42,12 @@
 
   BaseBatchReader(
       Table table,
+      String branch,
```

Comment:
Are we sure it is a good idea to pass `branch` all the way to readers? It feels like readers should be concerned about the schema to read (or snapshot schema) but not about branches.

Reply:
Seems like we only use it get table schema for the branch. I'd consider either passing `tableSchema` or creating a special `Table` to hide away branching logic and return the branch schema for `table.schema()`.

Reply:
It is just an idea, though. Let's discuss and let me know if there is a good reason to pass `branch` all the way to readers.

Reply:
It feels a bit awkward to implement a `TableWithBranch extends Table` and vend schema through `table.schema()` just to avoid changing constructors, because it is not a table and does not have all the other capabilities.

I decided to pass in a `String branch` instead of `Schema snapshotSchema` just because it seems simpler, but the other way would also work, but will also pass schema all the way to all readers with mostly the same amount of changes. This might be fine though, would like to know what others think as well.

Reply:
I thought about this a bit more, I think passing in schema might be better. The schema is used only for the delete filter, so we can call it `deleteSchema`

Reply:
Actually maybe `tableSchema` is better, in case it will be used for other prupose. It also matches the name in 

```
protected DeleteFilter(
      String filePath,
      List<DeleteFile> deletes,
      Schema tableSchema,
      Schema requestedSchema,
      DeleteCounter counter) {
```

Reply:
I'm fine passing either the table schema or the branch. I looked into this as well and I thought that the advantage of passing the branch was that we preserve all of the information needed whenever we pass table. If we pass table without the branch then we'll possibly pass a table and a schema that doesn't match it's current state.